### PR TITLE
[runbook-sync-downstream] Close outdated PRs

### PR DESCRIPTION
**Pull request by machadovilaca**
**What this PR does / why we need it**:

When new changes are added to the runbooks, a new PR is created upstream updating all diffs between runbooks. But an old PR that has not been merged yet would remain open with only a portion of the diffs. This change makes the tool close those outdated PRs.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
none
```

### CODE REVIEW

**Comment by sradco** [comment]
Why is the existing PR not updated with the new changes?

**Comment by machadovilaca** [comment]
> Why is the existing PR not updated with the new changes?

We would have to easy way in the next runs to know if those changes are already applied or not.
Currently we track the branch dates to know which changes are already proposed downstream, if we updated old PRs, we would need to come up with a more complicate mechanism to iterate all the PRs (open and closed) and check each commit to see if new changes are present there

File: tools/runbook-sync-downstream/main.go [hunk 1]
```diff
@@ -293,16 +297,56 @@ func (rbSync *runbookSync) createPR(branchName string, title string, body string
...
+
+func (rbSync *runbookSync) closeOutdatedRunbookPR(oldPR *github.PullRequest, newPr *github.PullRequest) error {
+	klog.Infof("closing outdated PR: %s", oldPR.GetHTMLURL())
+
+	if rbSync.dryRun {
+		klog.Warning("[DRY RUN] skipping PR close")
+		return nil
+	}
+
+	body := *oldPR.Body + fmt.Sprintf("\n\nClosed in favor of %s.", newPr.GetHTMLURL())
```
Comment (avlitman) [inline tools/runbook-sync-downstream/main.go:341]:
```suggestion
	body := fmt.Sprintf("%s\n\nThis pull request has been closed in favor of a newer one. Please refer to the updated PR for the latest changes and discussion: %s", 
    *oldPR.Body, newPr.GetHTMLURL())
```

Comment (machadovilaca) [reply tools/runbook-sync-downstream/main.go]: updated

**Pull request merged**